src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -536,6 +536,7 @@ export function useSwapInputsController({
 
   const onChangedPercentage = useDebouncedCallback(
     (percentage: number) => {
+      if (percentageToSwap.value === percentage) return;
```

Comment:
Since `percentageToSwap.value` is derived from `sliderXPosition.value`, which is set immediately when you drag the slider, and especially because this function is debounced, won’t this prevent execution of this function when it should in fact execute?

What specific case is this trying to guard against?

There are similar checks in the slider component itself but perhaps they need adjusting.